core/io/resource_saver.cpp

```diff
@@ -158,7 +158,7 @@ void ResourceSaver::get_recognized_extensions(const RES &p_resource, List<String
 
 void ResourceSaver::add_resource_format_saver(Ref<ResourceFormatSaver> p_format_saver, bool p_at_front) {
 
-	ERR_FAIL_COND(p_format_saver.is_null());
+	ERR_FAIL_COND_MSG(p_format_saver.is_null(), "Format saver cannot be empty.");
```

Comment:
This is not empty, but null, i.e. it's not a reference to a valid `ResourceFormatSaver` object. It might be `NULL`, or a `Ref<Potato2D>` :)

Same issue further up with null checks on `Ref<Image>`, it's not about the image being empty (sounds like a 0x0 px PNG), but the reference not pointing to a valid `Image` resource.